addons/l10n_vn_edi_viettel/models/account_move.py

```diff
@@ -0,0 +1,766 @@
+        self._l10n_vn_edi_add_seller_information(json_values)
+        self._l10n_vn_edi_add_payment_information(json_values)
+        self._l10n_vn_edi_add_item_information(json_values)
+        self._l10n_vn_edi_add_tax_breakdowns(json_values)
+        return json_values
+
+    def _l10n_vn_edi_add_general_invoice_information(self, json_values):
+        """ General invoice information, such as the model number, invoice symbol, type, date of issues, ... """
+        self.ensure_one()
+        invoice_data = {
+            'transactionUuid': str(uuid.uuid4()),
+            'invoiceType': self.l10n_vn_edi_invoice_symbol.invoice_template_id.template_invoice_type,
+            'templateCode': self.l10n_vn_edi_invoice_symbol.invoice_template_id.name,
+            'invoiceSeries': self.l10n_vn_edi_invoice_symbol.name,
+            # This timestamp is important as it is used to check the chronological order of Invoice Numbers.
+            # Since this xml is generated upon posting, just like the invoice number, using now() should keep that order
+            # correct in most case.
+            'invoiceIssuedDate': self._l10n_vn_edi_format_date(self.l10n_vn_edi_issue_date),
+            'currencyCode': self.currency_id.name,
+            'adjustmentType': '1',  # 1 for original invoice, which is the case during first issuance.
+            'paymentStatus': self.payment_state in {'in_payment', 'paid'},
+            'cusGetInvoiceRight': True,  # Set to true, allowing the customer to see the invoice.
+            'validation': 1,  # Set to 1, SInvoice will validate tax information while processing the invoice.
+        }
+
+        # When invoicing in a foreign currency, we need to provide the rate, or it will default to 1.
+        if self.currency_id.name != 'VND':
+            invoice_data['exchangeRate'] = self.env['res.currency']._get_conversion_rate(
+                from_currency=self.company_currency_id,
+                to_currency=self.currency_id,
+                company=self.company_id,
+                date=self.invoice_date or self.date,
+            )
+
+        adjustment_origin_invoice = None
+        if self.move_type == 'out_refund':  # Credit note are used to adjust an existing invoice
+            adjustment_origin_invoice = self.reversed_entry_id
+        elif self.l10n_vn_edi_replacement_origin:  # 'Reverse and create invoice' is used to issue a replacement invoice
+            adjustment_origin_invoice = self.l10n_vn_edi_replacement_origin
+
+        if adjustment_origin_invoice:
+            invoice_data.update({
+                'adjustmentType': '5' if self.move_type == 'out_refund' else '3',  # Adjustment or replacement
+                'adjustmentInvoiceType': self.l10n_vn_edi_adjustment_type or '',
+                'originalInvoiceId': adjustment_origin_invoice.l10n_vn_edi_invoice_number,
+                'originalInvoiceIssueDate': self._l10n_vn_edi_format_date(adjustment_origin_invoice.l10n_vn_edi_issue_date),
```

Comment:
is it ok if this field is in UTC?

Reply:
As what we actually set is the time converted into seconds since unix epoch, I would say it should be fine :thinking: 
From when I tested, the times seemed correct on the paltform